examples/linear_model/plot_ridge_coeffs.py

```diff
@@ -1,71 +1,88 @@
-are smaller (eventually converging at 0) leading to a
-simpler and biased solution.
-These dependencies can be observed on the left plot.
-
-The right plot shows the mean squared error between the
-coefficients found by the model and the chosen vector w.
-Less regularised models retrieve the exact
-coefficients (error is equal to 0), stronger regularised
-models increase the error.
-
-Please note that in this example the data is non-noisy, hence
-it is possible to extract the exact coefficients.
+When a model overfits, we want to regularize it's weights so it can generalize
+better on unseen data. Regularization reduces a model's reliance on specific
+information obtained from the training samples.
 
+This example illustrates how L2 regularization affects a model's performance by
+adding a penalty term to the Loss that increases with the coefficients (β).
+
+The regularized loss function is given by:
+
+RegularizedLoss = Loss(X, y, β) + Penalty(β)
+
+where:
+
+- X is the input data,
+
+- y is the target variable,
+
+- β is the coefficients or weights associated with the features.
+
+In Ridge regularization, the penalty term is proportional to the sum of the
+squares of the coefficients. In effect, it discourages any single coefficient βi
+from taking an excessively large value. This indirectly influences the
+individual coefficients towards smaller values, forcing the model to shrink
+certain coefficients or even select fewer features. Thus, Ridge regularization
+is promoting model simplicity or feature selection to prevent overfitting.
 """
 
 # Author: Kornel Kielczewski -- <kornel.k@plusnet.pl>
 
-import matplotlib.pyplot as plt
-import numpy as np
-
+# %%
+# Creating Dataset
+# ----------------
+# We make a toy dataset, that's suitable to detect regression. Please note that
+# in this example the data is non-noisy, hence it is possible to extract the
+# exact coefficients.
 from sklearn.datasets import make_regression
-from sklearn.linear_model import Ridge
-from sklearn.metrics import mean_squared_error
-
-clf = Ridge()
 
 X, y, w = make_regression(
     n_samples=10, n_features=10, coef=True, random_state=1, bias=3.5
 )
 
+# Obtain the true coefficients
+w
+
+# %%
+# Training Ridge Regressor
+# ------------------------
+# For a model, we use :class:`Ridge`, a linear model with L2 regularization. The
+# model parameter `alpha` is a postive constant that multiplies the penalty
+# term, controlling the regularization strength.
+import numpy as np
+from sklearn.linear_model import Ridge
+from sklearn.metrics import mean_squared_error
+
+clf = Ridge()
+
+# Generate values for `alpha` that are evenly distributed on a logarithmic scale
+alphas = np.logspace(-3, 3, 200)
 coefs = []
 errors = []
 
-alphas = np.logspace(-6, 6, 200)
-
 # Train the model with different regularisation strengths
 for a in alphas:
     clf.set_params(alpha=a)
     clf.fit(X, y)
     coefs.append(clf.coef_)
     errors.append(mean_squared_error(clf.coef_, w))
```

Comment:
I am thinking that we could have a `GridSearchCV` instead of doing things by hand but I am not sure about the plot that we will get. But it would provide some estimate of the variance for the coefficients and errors.

Reply:
Uhm actually `GridSearchCV` does not allow storing models. I confused it with `cross_validate`. This could also be an alternative.

Reply:
Let's keep the example as it was with a single split.